torch/onnx/symbolic_opset11.py

```diff
@@ -648,29 +648,20 @@ def im2col(g, input, kernel_size, dilation, padding, stride):
 @parse_args('v', 'i', 'i')
 def flatten(g, input, start_dim, end_dim):
     dim = input.type().dim()
+    if dim is None:
+        return _unimplemented("dim",
+                              "ONNX and PyTorch use different strategies to split the input. "
+                              "Input rank must be known at export time.")
+
     # use ONNX's Flatten operator for cases where the output shape is 2D
     if start_dim == 1:
         if (end_dim == -1 or (end_dim is not None and end_dim == dim - 1)):
```

Comment:
For this case, isn't it ok if `dim` is `None`? 
Same question for opset9.

Reply:
I don't think end_dim = None-1 is valid statement

Reply:
I don't think it is valid either. But if `end_dim` is -1 or `None`, then can't it be allowed for `dim` to be `None`?

Reply:
I don't think end_dim can be None, if not specified default value is -1. I don't think this line is correct, maybe the original author meant dim is not None instead end_dim.

Reply:
I think `dim is not None` makes more sense than `end_dim is not None`. Should we change it?
And if so, the check above that returns `unimplemented(...)` should be done after this if.

Reply:
I just removed it. The above check for dim should cover it and other cases as well.